File: RESTful Samples/Ruby/AdoDataSource.rb
# Copyright (c) 2015 Windward Studios
#
#
# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), 
# to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software,
#  and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
#
# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
#
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, 
# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, 
# WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.

require 'rexml/document'
require_relative 'DataSource'
include REXML

class AdoDataSource < DataSource
	@className
	@connectionString
	
	def initialize(className, connectionString)
		@className = className
		@connectionString = connectionString
	end
	
	def GetXml(name)
		element = Element.new "Datasource"
		e = element.add_element "Name" 
		e.add_text name.to_s
		e = element.add_element "Type" 
		e.add_text "sql"
		e = element.add_element "ClassName" 
		e.add_text @className.to_s
		e = element.add_element "ConnectionString" 
		e.add_text @connectionString.to_s
		if @Variables != nil && @Variables.length > 0
			element.add_element(GetVariablesXml)
		end
		return element
	end	
end	

File: RESTful Samples/RestfulEngine_JavaClient/src/client/AdoDataSource.java
package client;

import org.w3c.dom.Document;
import org.w3c.dom.Element;

import java.io.IOException;

/**
 * Created by Bassem on 4/14/2015.
 */
public class AdoDataSource extends DataSource {
    private String className;
    private String connectionString;

    public AdoDataSource(String className, String connectionString)
    {
        this.className = className;
        this.connectionString = connectionString;
    }
    @Override
    public Element getXml(String name, Document doc) throws IOException {
        Element dataSourceElement = doc.createElement("Datasource");

        Element nameElement = doc.createElement("Name");
        nameElement.appendChild(doc.createTextNode(name));
        dataSourceElement.appendChild(nameElement);

        Element typeElement = doc.createElement("Type");
        typeElement.appendChild(doc.createTextNode("sql"));
        dataSourceElement.appendChild(typeElement);

        Element classNameElement = doc.createElement("ClassName");
        classNameElement.appendChild(doc.createTextNode(className));
        dataSourceElement.appendChild(classNameElement);

        Element conStgElement = doc.createElement("ConnectionString");
        conStgElement.appendChild(doc.createTextNode(connectionString));
        dataSourceElement.appendChild(conStgElement);

        if(variables != null && variables.size() > 0){
            dataSourceElement.appendChild(getVariablesXml(doc));
        }

        return dataSourceElement;
    }
}
